File: backend/src/main/java/com/izonescraper/sunwayizonescraper/AI/StudySlot.java
package com.izonescraper.sunwayizonescraper.AI;

import com.fasterxml.jackson.annotation.JsonProperty;

public record StudySlot(@JsonProperty("date") String date,
                        @JsonProperty("subject") String subject,
                        @JsonProperty("time") String time) {
}

File: frontend/lib/models/study_slot.dart
class StudySlot {
  final String date;
  final String subject;
  final String time;

  const StudySlot({
    required this.date,
    required this.subject,
    required this.time,
  });

  factory StudySlot.fromJson(
    Map<String, dynamic> json,
  ) {
    return StudySlot(
      date: json['date'] ?? '',

      subject: json['subject'] ?? '',

      time: json['time'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'date': date,

      'subject': subject,

      'time': time,
    };
  }
}
